ocs_ci/ocs/platform_nodes.py

```diff
@@ -3459,3 +3470,109 @@ def wait_for_nodes_to_stop_or_terminate(self, nodes):
         raise NotImplementedError(
             "wait for nodes to stop or terminate functionality is not implemented"
         )
+
+
+class ROSAHCPNode(AWSNodes):
+    """
+    ROSA HCP Nodes class
+
+    """
+
+    def __init__(self, node_conf=None, node_type=None):
+        self.node_conf = node_conf
+        self.node_type = node_type
+        super(ROSAHCPNode, self).__init__()
+
+    def create_and_attach_nodes_to_cluster(
+        self, node_conf, node_type=constants.RHCOS, num_nodes=None
+    ):
+        """
+        Create nodes and attach them to cluster
+        Use this function if you want to do both creation/attachment in a single call
+        (attachment is skipped due to AWSNodes logic for non-"upi" deployment)
+
+        Args:
+            node_conf (dict): node configuration - custom dict that we use in rosa create/edit machinepool
+            e.g.
+            {'machinepool': '<mp_name>',
+            'instance_type': '<m5.xlarge>',
+            'subnet': '<subnet_id>',
+            'availability_zone': '<az>',
+            'disk_size': <300Gb def for m5.xlarge>,
+            'enable_autoscaling': <False by default>,
+            'labels': '<coma separated string of key=value>',
+            'multi-availability-zone': <False by default>,
+            'tags': '<coma separated string of key=value:ScheduleType>',
+            }
+            node_type (str): type of node to be created
+            num_nodes (int): Number of node instances to be created
+
+        """
+        if node_type != constants.RHCOS:
+            raise UnsupportedOSType(
+                "Only RHCOS is supported for ROSA HCP. Check up "
+                "https://docs.redhat.com/en/documentation/red_hat_openshift_service_on_aws/4/html/"
+                "tutorials/getting-started-with-rosa#underlying-node-operating-system"
+            )
+        if num_nodes is None:
+            raise ValueError("num_nodes must be provided")
```

Comment:
I am still determining the exception here. We don't expect to provide a value when we give a default param. Maybe we should give `1` as the default value, or we can mention it in the docstring.

Reply:
So, will the default value `1` not work? If not, we can keep it as it is now.

Reply:
I will think about it; I will continue work on it an already have another bunch of commits that I'd want to merge separately. I will include it there.